workflows/amr/run.wdl

```diff
@@ -625,15 +625,18 @@ task tsvToSam {
         contigs_fasta = pysam.Fastafile("~{contigs}")
 
         # Load columns of interest from CSV and drop rows with at least one NaN
-        df = pd.read_csv("~{final_summary}", sep="\t", usecols=[COLUMN_GENE_ID, COLUMN_CONTIG_NAME]).dropna()
+        df = pd.read_csv("~{final_summary}", sep="\t", usecols=[COLUMN_GENE_ID, COLUMN_CONTIG_NAME])
+
+        # Create BAM with mock reference lengths for the header (do this before df.dropna() so we
+        # list all gene IDs in the SAM header). If no gene IDs are found at all, have a mock gene
+        # to make sure we can create the SAM file with no errors (web app will look for that file)
+        gene_ids = df[COLUMN_GENE_ID].dropna().unique().tolist() or ["NoGenes"]
```

Comment:
This means the SAM header will now contain all the geneIDs mentioned in the TSV file, not just the ones associated with contigs.

The `or ["NoGenes"]` is to support cases where no gene IDs are found in the file at all. If that ever happens, the SAM file will still be created without error.